RecoTracker/LSTCore/src/alpaka/NeuralNetwork.h

```diff
@@ -136,6 +137,44 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE::lst {
     }
   }  // namespace t3dnn
 
+  namespace tcdnn {
+
+    template <typename TAcc>
+    ALPAKA_FN_ACC ALPAKA_FN_INLINE void runInference(TAcc const& acc,
+                                                     const float tc_pt,
+                                                     const float tc_eta,
+                                                     const float tc_phi,
+                                                     const float tc_type,
+                                                     float (&embedding)[4]) {
```

Comment:
is 4 outputs for 3 inputs actually practical?
IIUC, `tc_type` here is the same 8

Reply:
This method works both across track types and within a given track type. The DNN I trained for embeddings here is for arbitrary track type, although I have not implemented that yet in the cleanings. I will explain more my plans for this on Friday.